src/fusiontensor/fusiontensoraxes.jl

```diff
@@ -0,0 +1,137 @@
+  outer_axes::BT
+
+  function FusionTensorAxes{BT}(bt) where {BT}
+    @assert BT === typeof(promote_sectors(bt))
+    return new{BT}(bt)
+  end
+end
+
+# =====================================  Accessors  ========================================
+
+outer_axes(fta::FusionTensorAxes) = fta.outer_axes
+
+# ====================================  Constructors  ======================================
+
+function FusionTensorAxes(bt::BlockedTuple{2})
+  promoted = promote_sectors(bt)
+  return FusionTensorAxes{typeof(promoted)}(promoted)
+end
+
+function FusionTensorAxes(codomain_legs, domain_legs)
+  return FusionTensorAxes(tuplemortar((codomain_legs, domain_legs)))
+end
+
+# ==================================  Base interface  ======================================
+
+for f in [
+  :(broadcastable), :(Tuple), :(axes), :(firstindex), :(lastindex), :(iterate), :(length)
+]
+  @eval Base.$f(fta::FusionTensorAxes) = Base.$f(outer_axes(fta))
+end
+
+for f in [:(getindex), :(iterate)]
+  @eval Base.$f(fta::FusionTensorAxes, i) = $f(outer_axes(fta), i)
+end
+
+function Base.getindex(fta::FusionTensorAxes, bp::AbstractBlockPermutation)
+  return FusionTensorAxes(outer_axes(fta)[bp])
+end
+
+Base.copy(fta::FusionTensorAxes) = FusionTensorAxes(copy.(outer_axes(fta)))
+
+Base.deepcopy(fta::FusionTensorAxes) = FusionTensorAxes(deepcopy.(outer_axes(fta)))
+
+function Base.:(==)(a::FusionTensorAxes, b::FusionTensorAxes)
+  blocklengths(a) != blocklengths(b) && return false
+  for i in 1:length(a)
+    !space_isequal(a[i], b[i]) && return false
+  end
+  return true
+end
+
+# ================================  BlockArrays interface  =================================
+
+for f in [:(blocklength), :(blocklengths), :(blocks)]
+  @eval BlockArrays.$f(fta::FusionTensorAxes) = $f(outer_axes(fta))
+end
+
+# ==============================  GradedArrays interface  ==================================
+
+function GradedArrays.sector_type(
+  ::Type{FTA}
+) where {BT<:BlockedTuple{2,(0, 0)},FTA<:FusionTensorAxes{BT}}
+  return TrivialSector
+end
+
+function GradedArrays.sector_type(::Type{FTA}) where {BT,FTA<:FusionTensorAxes{BT}}
+  return sector_type(type_parameters(type_parameters(BT, 3), 1))
+end
+
+# ==============================  TensorAlgebra interface  =================================
+
+TensorAlgebra.trivial_axis(fta::FusionTensorAxes) = trivial_axis(sector_type(fta))
+
+# ==============================  FusionTensor interface  ==================================
+
+codomain_axes(fta::FusionTensorAxes) = fta[Block(1)]
+
+domain_axes(fta::FusionTensorAxes) = fta[Block(2)]
+
+function codomain_axis(fta::FusionTensorAxes)
+  if ndims_codomain(fta) == 0
+    return trivial_axis(fta)
+  end
+  return ⊗(codomain_axes(fta)...)
+end
+
+function domain_axis(fta::FusionTensorAxes)
+  if ndims_domain(fta) == 0
+    return dual(trivial_axis(fta))
+  end
+  return dual(⊗(dual.(domain_axes(fta))...))
+end
+
+ndims_codomain(fta::FusionTensorAxes) = length(codomain_axes(fta))
```

Comment:
```suggestion
length_codomain(fta::FusionTensorAxes) = length(codomain(fta))
```
Similar to the comment above.